pkg/inspector/image-inspector.go

```diff
@@ -47,11 +40,30 @@ type ImageInspector interface {
 // defaultImageInspector is the default implementation of ImageInspector.
 type defaultImageInspector struct {
 	opts iicmd.ImageInspectorOptions
+	// an optional image server that will server content for inspection.
+	imageServer apiserver.ImageServer
 }
 
 // NewDefaultImageInspector provides a new default inspector.
 func NewDefaultImageInspector(opts iicmd.ImageInspectorOptions) ImageInspector {
-	return &defaultImageInspector{opts}
+	inspector := &defaultImageInspector{
+		opts: opts,
+	}
+
+	// if serving then set up an image server
+	if len(opts.Serve) > 0 {
+		imageServerOpts := apiserver.ImageServerOptions{
+			ServePath:     opts.Serve,
+			HealthzURL:    HEALTHZ_URL_PATH,
+			APIURL:        API_URL_PREFIX,
+			APIVersions:   apiserver.APIVersions{Versions: []string{VERSION_TAG}},
+			MetadataURL:   METADATA_URL_PATH,
+			ContentURL:    CONTENT_URL_PREFIX,
+			ImageServeURL: opts.DstPath,
```

Comment:
actually, this won't work here, need to set it later since we manipulate the `DstPath` later for tmpfs if it's empty.  Will fix up